### PR TITLE
Shorten timeout duration for environment close

### PR DESCRIPTION
**Pull request by harperj**
### Proposed change(s)

The timeout duration for closing an environment was set to the
same duration as the timeout when waiting for a response from the
still-running environment.  This led to long waits for the error
response when communication version wasn't matching.

This change updates the timeout duration so that we wait a fixed
timeout of 5 seconds before force-killing the environment worker.

### Types of change(s)

- [ ] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [x] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Comment by chriselion** [comment]
This looks good as it is, but just wanted to give one other possible approach:
* make _close() take a timeout parameter. This can default to None and use self.timeout_wait if the actual value is None
* pass 0 timeout in the cases we want to close immediately (or make _close_now() which calls _close() with 0 timeout). I think this should happen everywhere except for `close()` (so mismatched API, connection timeout, and invalid launch_string all close immediately)
* Keep `close()` using self.timeout_wait

**Comment by harperj** [comment]
I like that solution @chriselion.  I've made the change.

File: ml-agents-envs/mlagents_envs/environment.py [hunk 1]
```diff
@@ -444,13 +444,15 @@ def close(self):
         else:
             raise UnityEnvironmentException("No Unity environment is loaded.")
 
-    def _close(self):
+    def _close(self, timeout: Optional[int] = None) -> None:
```
Comment (vincentpierre) [inline ml-agents-envs/mlagents_envs/environment.py]: There is some added logic here, can you add a docstring explaining what the argument does?

Comment (harperj) [reply ml-agents-envs/mlagents_envs/environment.py]: Done.

**Pull request merged**